### PR TITLE
Feature/calendar and statistics

### PR DESCRIPTION
**Pull request by tmaciulis22**
5 PRs in 1. Migrated not only statistics and calendar, but also face detection, sign and registration with face. TODO in next PR: add more charts and put emotion struct to DB.

### CODE REVIEW

File: app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterFaceActivity.cs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            }
+            else
+            {
+                ActivityCompat.RequestPermissions(this, new string[] { Manifest.Permission.Camera }, REQUEST_CAMERA);
+            }
+        }
+
+        private void StartCamera()
+        {
+
```
Comment (AArlauskas) [inline app-android-xamarin/Moodis/Moodis/Feature/Register/RegisterFaceActivity.cs]: is this an empty line 😮

File: app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs [hunk 2]
```diff
@@ -9,14 +9,18 @@
...
+using Android.Views;
 
 namespace Moodis.Feature.SignIn
 {
     [Activity(Label = "Sign In")]
     public class SignInActivity : AppCompatActivity
     {
         public static int REQUEST_CODE_REGISTER = 1;
-        private SignInViewModel SignInViewModel = new SignInViewModel();
+        public static int REQUEST_CODE_FACE = 1;
```
Comment (AArlauskas) [inline app-android-xamarin/Moodis/Moodis/Feature/SignIn/SignInActivity.cs]: A tu man kartais nepasakojai kad cia skaiciukas dva turetu buti?

**Review by AArlauskas — APPROVED** [review]
Too long didn't read

File: app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs [hunk 3]
```diff
@@ -69,7 +73,7 @@ private async Task<IList<DetectedFace>> DetectFaceEmotions(string imageFilePath)
             }
             catch (APIErrorException apiException)
             {
-                Console.WriteLine(API_ERROR + " " + apiException.StackTrace);
+                Console.WriteLine(API_ERROR + " " + apiException.StackTrace);//TODO change to Log
```
Comment (Pilkas-git) [inline app-android-xamarin/Moodis/Moodis/Network/Face/Face.cs]: change to log

File: app-android-xamarin/Moodis/Moodis/Models/ImageInfo.cs [hunk 4]
```diff
@@ -30,16 +32,18 @@ public int CompareTo(object obj)
             }
         }
 
+        [PrimaryKey, AutoIncrement]
+        public int Id { get; set; }
+        public int UserId { get; set; }
         public string ImagePath { get; set; }
-        public Emotion[] emotions;
+        public List<Emotion> emotions;
```
Comment (Pilkas-git) [inline app-android-xamarin/Moodis/Moodis/Models/ImageInfo.cs:39]: Just a question, does the db not support structs ?

File: app-android-xamarin/Moodis/Moodis/Feature/CameraFeature/CameraFragment.cs [hunk 5]
```diff
@@ -38,12 +47,46 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
             return view;
         }
 
+        public override async void OnViewCreated(View view, Bundle savedInstanceState)
```
Comment (Pilkas-git) [inline app-android-xamarin/Moodis/Moodis/Feature/CameraFeature/CameraFragment.cs]: Don't think this needs async as you only await on line 65 which is already handled by AfterTakenPicture = async (sender, e) =>

**Review by Pilkas-git — CHANGES_REQUESTED** [review]

File: app-android-xamarin/Moodis/Moodis/Models/ImageInfo.cs (hunk 4 above)
Comment (tmaciulis22) [reply app-android-xamarin/Moodis/Moodis/Models/ImageInfo.cs:39]: Will solve this issue in next PR

**Review by Pilkas-git — APPROVED** [review]
Leave some work for the rest of us :DD

**Pull request merged**